crates/fuel-core/src/database.rs

```diff
@@ -411,6 +415,65 @@ impl fuel_core_storage::vm_storage::VmStorageRequirements for Database {
     }
 }
 
+impl AtomicView for Database {
+    type View = Database;
+
+    type Height = BlockHeight;
+
+    fn latest_height(&self) -> BlockHeight {
+        // TODO: The database should track the latest height inside of the database object
+        //  instead of fetching it from the `FuelBlocks` table. As a temporary solution,
+        //  fetch it from the table for now.
+        self.latest_height().unwrap_or_default()
+    }
+
+    fn view_at(&self, _: &BlockHeight) -> StorageResult<Self::View> {
+        // TODO: Unimplemented until of the https://github.com/FuelLabs/fuel-core/issues/451
```

Comment:
It looks like this is merged now.

Reply:
Without state rewind we can't implement it properly=) It uses the latest for now to simulate the current behavior=)